src/test/java/com/github/psambit9791/jdsp/TestShortTimeFourier.java

```diff
@@ -404,4 +408,48 @@ public void testTimeAxis2() {
 
         Assertions.assertArrayEquals(result, out, 0.001);
     }
+
+    @Test
+    public void testSTFTLongSignal() throws IOException {
```

Comment:
Due to the nature of how FFT works in Apache Math3, we might need to pad the signals with 0 to the nearest power of 2. The outputs from the FFT are in powers of 2 whih probably will not map into the 5400 length. So, I am fairly sure, even if the STFT works properly as expected, due to the length difference, the test will fail.

Any chance you could check the newly added test under TestHilbert and implement something along those lines?

Reply:
Should the user get an error message when the frameLength or fourierLength isn't a power of 2 then? Because STFT fills in the output depending on the given frame length, but the FFT may return an output with greater dimensions than the given frameLength, because the Fourier-implementation already implements the zero-padding

Reply:
For STFT, all frames have the same length so the FFT will pad by the same length as well, which may returning a different length than DFT but which still remains constant between frames.

_Fourier also has a function called getSignalLength() [returns the length after padding] which can help determine the output dimensions.

So, I think not allowing users to have a padding length outside a power of 2 will be very limiting. Instead, handling it at the FFT end is neater.

Reply:
Yeah, but I find it counter-intuitive to specify a frame/Fourierlength and then having the STFT returning different sized frames, no?

Reply:
Technically we are analysing the same period because of the zero padding. Limiting the user to only window lengths in powers of two significantly reduces the periods of analysis that users can choose. As long as it is documented that the length of the FFT may not return the same size output, it should not be a problem because the key difference is the resolution, not the actual value.However it provides users with more liberty in selecting their periods of analysis.

Reply:
Yeah, maybe you're right. Would you mind trying to patch the testing and output for the STFT? I'm kinda cramped with other work at the moment

Reply:
Sure